+utils/compare_symbols.m

```diff
@@ -0,0 +1,149 @@
+this_dir = fileparts(mfilename('fullpath'));
+needed_symbols_file = fullfile(utils.Paths.toolbox_directory(), "source", "neuron_dllimports.h");
+
+if ismac
+    lib_ext = ".dylib";
+elseif isunix
+    lib_ext = ".so";
+else
+    lib_ext = "";
+end
+
+% Create the files with the lists of symbols in the libraries
+if optionals.compiled_symbols_file == ""
+    compiled_symbols_file = fullfile(this_dir, "neuronInterface.txt");
+    cmd = "nm '" + fullfile(utils.Paths.toolbox_lib_directory(), "neuronInterface" + lib_ext) + "' > '" + compiled_symbols_file + "'";
+    status = system(cmd);
+    if status
+        error("system call failed")
+    end
+else
+    compiled_symbols_file = optionals.compiled_symbols_file;
+end
+if optionals.nrniv_symbols_file == ""
+    nrniv_symbols_file = fullfile(this_dir, "nrniv.txt");
+    cmd = "nm '" + utils.Paths.libnrniv_file() + "' > '" + nrniv_symbols_file + "'";
+    status = system(cmd);
+    if status
+        error("system call failed")
+    end
+else
+    nrniv_symbols_file = optionals.nrniv_symbols_file;
+end
+
+% Get the symbols lists as one big char array
+% For easier later regex search, make sure there is a newline at the start and end
+compiled_content = [10 fileread(compiled_symbols_file) 10];
+nrniv_content = [10 fileread(nrniv_symbols_file) 10];
+
+%% Parse the needed_symbols_file
+% Assume that lines starting with MANGLED
+% and NON_MANGLED have a symbol that should be checked
+possible_symbols = readlines(needed_symbols_file);
+max_nr_symbols = length(possible_symbols);
+varTypes = ["string", "logical", "string", "logical", "string"];
+varNames = ["Symbol", "is_mangled", "Interface", "is_matching", "Nrniv"];
+list_symbols = table('Size', [max_nr_symbols, length(varNames)], 'VariableTypes', varTypes, 'VariableNames', varNames);
+for i = 1:max_nr_symbols
+    line = possible_symbols(i);
+    if contains(line, "Import functions for which name mangling goes awry")
+        % TODO: something for the functions that we know are problematic,
+        % for now they need to be manually checked
+        break
+    end
+    if line.startsWith({'MANGLED', 'NON_MANGLED'})
+        parts = line.split;
+        if length(parts) ~= 3
+            error("line in file starting with MANGLED cannot be split in exactly three parts, script needs to be adapted")
+        elseif ~parts.endsWith(';')
+            error("The third part of a line in file starting with MANGLED or NON_MANGLED does not end with ;, script needs to be adapted")
+        end
+        % symbols that are still in neuron_dllimports, but are no longer
+        % used anywhere. Can possibly make this an output of this script,
+        % instead of hardcoding here
+        % if any(contains(["nrn_popsec", "hoc_call_objfunc", "hoc_obj_ref", "hoc_ret", "hoc_objectdata"], parts{3}(1:end-1)))
+        %     continue
+        % end
+        list_symbols.Symbol(i) = parts{3}(1:end-1);
+        list_symbols.is_mangled(i) = line.startsWith('MANGLED');

```

Comment:
any idea why the first `'` here is being given a dark background?

Reply:
VS code does not agree that there's an issue here. Is it a GitLab syntax highlighter quirk? Does it go away if we use double quotes?